### PR TITLE
fix: fixed a bug where external urls could not be defined as refs within the async api

### PR DESCRIPTION
**Pull request by vlad-geiger-msg-systems**

### CODE REVIEW

File: sda-commons-shared-asyncapi/src/main/java/org/sdase/commons/shared/asyncapi/internal/JsonSchemaEmbedder.java [hunk 1]
```diff
@@ -135,4 +138,8 @@ private JsonNode ensureParentNode(JsonNode target, JsonPointer pointer) {
   private String replaceSpecialCharacters(String value) {
     return value.replaceAll("[^a-zA-Z0-9]", "");
   }
+
+  private boolean shouldEmbed(String value) {
+    return !value.startsWith("https://");
```
Comment (christopher-cudennec) [inline sda-commons-shared-asyncapi/src/main/java/org/sdase/commons/shared/asyncapi/internal/JsonSchemaEmbedder.java:143]: Is this the only case? Do we already know what kind of URLs we should ignore?

Comment (christopher-cudennec) [reply sda-commons-shared-asyncapi/src/main/java/org/sdase/commons/shared/asyncapi/internal/JsonSchemaEmbedder.java:143]: There doesn't seem to be a test for the "https" case, right? Could you please add one?

Comment (vlad-geiger-msg-systems) [reply sda-commons-shared-asyncapi/src/main/java/org/sdase/commons/shared/asyncapi/internal/JsonSchemaEmbedder.java:143]: there is one, please check https://github.com/SDA-SE/sda-dropwizard-commons/pull/1483/files#diff-a6cc53dae6615711add1a52b0ea763497936180c5e6538c440444c518b20af78R57 and https://github.com/SDA-SE/sda-dropwizard-commons/pull/1483/files#diff-8a06b4e84467bf2f3b7fe793b3eb83226f42663b016787c691a2f30be6f53e00R17

**Review by christopher-cudennec — COMMENTED** [review]
My last comment before approving:
Could you squash your commits and use prefix `fix(shared-asyncapi): ...` please?

**Review by christopher-cudennec — APPROVED** [review]

**Pull request merged**